aten/src/ATen/templates/StaticUnboxingWrappers.cpp

```diff
@@ -0,0 +1,118 @@
+#include "torch/csrc/jit/runtime/operator.h"
+#include "torch/csrc/jit/runtime/custom_operator.h"
+#include "torch/csrc/jit/runtime/register_ops_utils.h"
+
+#include "torch/csrc/autograd/profiler.h"
+#include "torch/csrc/autograd/generated/variable_factories.h"
+
+#include <ATen/ATen.h>
+#include <ATen/core/functional.h>
+#include <ATen/core/interned_strings.h>
+
+#include <algorithm>
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <sstream>
+#include <stdexcept>
+#include <tuple>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+// ${generated_comment}
+
+// NOTE [Sharded File]: This file is generated in a sharded fashion to speed up
+// incremental rebuilds. See the comment at the top of
+// templates/VariableType.cpp for an analogous, in-depth discussion.
+//
+// Note that unlike VariableType.cpp, when sharding this file we take
+// care to generate all overloads of a particular name in a single
+// file and in a particular order. See gen_jit_dispatch.py for
+// details.
+
+namespace torch { namespace jit {
```

Comment:
The generated code is kernel under the `torch::jit` namespace and there are a bunch of related jit headers. I think we should be able to slice stuff so that it lives in and only depends on ATen, since the current templated unboxing logic is in ATen (stuff like `IValue` is currently defined in ATen, at least). I also think it might be hard to stare at the PR to uncover any issues with swapping out the templated unboxing logic for this. The easiest way to avoid that would be to also get normal eager mode to use this code along with this PR (and I can help with that!)